### ISSUE TITLE
Bug: EventHub Schemaregistry missing SchemaType Json

### ISSUE THREAD

**Issue by okaerin** [Azure/azure-rest-api-specs#24417]
[SchemaRegistry](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventhub/resource-manager/Microsoft.EventHub/stable/2021-11-01/SchemaRegistry.json#L278) from the EventHub is missing a schema type `JSON` in the enumeration.

**Comment by navba-MSFT** [comment]
@okaerin Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@okaerin The JSON scheme type is still in preview. see [here](https://learn.microsoft.com/en-us/azure/event-hubs/schema-registry-concepts#json-schema-preview) and [here](https://learn.microsoft.com/en-us/azure/event-hubs/create-schema-registry#:~:text=JSON%20serialization%20format%20is%20also%20supported%20(preview).).

The REST APi specs **stable version** is from 2021-11-01:
https://github.com/Azure/azure-rest-api-specs/blob/ab84b777992cf4ca170a18e1b8e5f3e437209888/specification/eventhub/resource-manager/Microsoft.EventHub/stable/2021-11-01/SchemaRegistry.json#L278

The last preview version we have is from 2022-10-01-preview.

I am checking this further with the Product owners on when will Json schema type be included in the preview api version ? I will keep you posted.

**Comment by okaerin** [comment]
@navba-MSFT That is true, but it is also missing in the preview, and I confirmed with a rest call (using the [try it function](https://learn.microsoft.com/en-us/rest/api/eventhub/preview/schema-registry/create-or-update?tabs=HTTP#request-body))that the preview API supports `JSON` already. So it is missing from 2022-10-01-preview: https://github.com/Azure/azure-rest-api-specs/blob/ab84b777992cf4ca170a18e1b8e5f3e437209888/specification/eventhub/resource-manager/Microsoft.EventHub/preview/2022-10-01-preview/SchemaRegistry.json#L278
It would be good to know, as our project would like to rely on IaC by terraform, which relys on a spec update from this repo.

**Comment by navba-MSFT** [comment]
@okaerin Thanks for your reply. I am working internally on this with the Product Owners. I will keep you posted.

**Comment by navba-MSFT** [comment]
@okaerin Apologies for the late reply. I have heard back from the Product Owners. We are in the process of updating the 2022-10-01-preview spec to include JSON schema format. We’ll update this thread once it is done.

Is this blocking you from using JSON schema format in your use case? Awaiting your reply.

**Comment by okaerin** [comment]
@navba-MSFT Thank you for looking into this matter. I am glad to hear that it is in the pipeline.

It blocks me because we can't finalize our infrastructure as code because Terraform relies on your API specification to generate their code.
